pallets/proxy-teleport/src/lib.rs

```diff
@@ -0,0 +1,206 @@
+		Instruction::{
+			BuyExecution, DepositAsset, DepositReserveAsset, InitiateReserveWithdraw,
+			ReceiveTeleportedAsset, Transact, WithdrawAsset,
+		},
+		MultiAsset, MultiAssetFilter, MultiAssets, Parent, SendXcm, WeightLimit, WildMultiAsset,
+		Xcm, XcmHash,
+	},
+	VersionedMultiAssets, VersionedMultiLocation, VersionedResponse, VersionedXcm,
+};
+
+// #[cfg(test)]
+// mod mock;
+
+// #[cfg(test)]
+// mod tests;
+
+// #[cfg(feature = "runtime-benchmarks")]
+// mod benchmarking;
+// pub mod weights;
+// pub use weights::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + pallet_xcm::Config {
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// An error ocured during send
+		SendError,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Execution of an XCM message was attempted.
+		Attempted { outcome: xcm::latest::Outcome },
+		/// A XCM message was sent.
+		Sent {
+			origin: MultiLocation,
+			destination: MultiLocation,
+			message: Xcm<()>,
+			message_id: XcmHash,
+		},
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::call_index(0)]
+		#[pallet::weight(10_000)]
+		pub fn proxy_native_teleport(
+			origin: OriginFor<T>,
+			dest: Box<VersionedMultiLocation>,
+			beneficiary: Box<VersionedMultiLocation>,
+			native_asset_amount: u128,
+			proxy_asset: Box<VersionedMultiAssets>,
+			fee_asset_item: u32,
+		) -> DispatchResult {
+			Self::do_proxy_teleport_assets(
+				origin,
+				dest,
+				beneficiary,
+				native_asset_amount,
+				proxy_asset,
+				fee_asset_item,
+			)
+		}
+	}
+}
+
+impl<T: Config> Pallet<T> {
+	fn do_proxy_teleport_assets(
+		origin: OriginFor<T>,
+		dest: Box<VersionedMultiLocation>,
+		beneficiary: Box<VersionedMultiLocation>,
+		native_asset_amount: u128,
+		proxy_asset: Box<VersionedMultiAssets>,
+		fee_asset_item: u32,
+	) -> DispatchResult {
+		//Unbox origin, destination and beneficiary.
+		let origin_location = T::ExecuteXcmOrigin::ensure_origin(origin)?;
+		let dest: MultiLocation =
+			(*dest).try_into().map_err(|()| pallet_xcm::Error::<T>::BadVersion)?;
+		let beneficiary: MultiLocation =
+			(*beneficiary).try_into().map_err(|()| pallet_xcm::Error::<T>::BadVersion)?;
+
+		//Create assets
+
+		// Native from local perspective
+		let native_asset = MultiAsset {
+			id: AssetId::Concrete(MultiLocation::here()),
+			fun: Fungibility::Fungible(native_asset_amount),
+		};
+
+		let assets = MultiAssets::from(vec![native_asset]);
+
+		// Native from foreign perspective
+		//TODO: Replace ID with parameter
+		let localtion_as_foreign =
+			MultiLocation { parents: 1, interior: Junctions::X1(Junction::Parachain(1836)) };
```

Comment:
Is it possible to do here as pallet XCM does so we don't need to hardcode the multi location of the native asset from the ecosystem perspective?


Also, there is a typo with 'location" in the variable name.

Reply:
Done! Please resolve if agreed.